popup.js

```diff
@@ -690,44 +657,30 @@ Asana.update(UserTypeahead.prototype, {
     return node;
   },
 
-  /**
-   * Generates a regular expression that will match strings which contain words
-   * that start with the words in filter_text. The matching is case-insensitive
-   * and the matching words do not need to be consecutive but they must be in
-   * the same order as those in filter_text.
-   *
-   * @param filter_text {String|null} The input text used to generate the regular
-   *  expression.
-   * @returns {Regexp}
-   */
-  _regexpFromFilterText: function(filter_text) {
-    if (!filter_text || filter_text.trim() === '') {
-      return null;
-    } else {
-      var escaped_filter_text = RegExp.escape(
-          filter_text.trim(),
-          /*opt_do_not_escape_spaces=*/true);
-      var parts = escaped_filter_text.trim().split(/\s+/).map(function(word) {
-        return "(" + word + ")";
-      }).join("(.*\\s+)");
-      return new RegExp("(?:\\b|^|(?=\\W))" + parts, "i");
-    }
-  },
-
   _confirmSelection: function() {
     this.user_id_to_select = this.selected_user_id;
   },
 
-  _updateFilteredUsers: function() {
-    var regexp = this._regexpFromFilterText(this.input.val());
-    this.filtered_users = this.users.filter(function(user) {
-      if (regexp !== null) {
-        var parts = user.name.split(regexp);
-        return parts.length > 1;
-      } else {
-        return user.name.trim() !== "";  // no filter
-      }
-    });
+  _updateUsers: function() {
+    var me = this;
+
+    this._request_counter += 1;
+    var current_request_counter = this._request_counter;
+    Asana.ServerModel.userTypeAhead(
+      Popup.options.default_workspace_id,
+      this.input.val(),
+      function (users) {
+        // Only update the list if no future requests have been initiated.
+        if (me._request_counter == current_request_counter) {
```

Comment:
Note we *always* use `===` in javascript, because the implicit conversions are inconsistent, make the code less explicit, and can mask subtle bugs.